ssl/statem/statem_srvr.c

```diff
@@ -1924,15 +1924,24 @@ static int tls_early_post_process_client_hello(SSL_CONNECTION *s)
         OSSL_TRACE_END(TLS_CIPHER);
     }
 
-    for (loop = 0; loop < clienthello->compressions_len; loop++) {
-        if (clienthello->compressions[loop] == 0)
-            break;
-    }
+    if (SSL_CONNECTION_IS_TLS13(s)) {
+        if (clienthello->compressions_len != 1 ||
```

Comment:
compression methods are defined as `opaque legacy_compression_methods<1..2^8-1>;` which means sending an empty list is a `decode_error`, not an `illegal_parameter`